### PR TITLE
feat: v0.1 SDK -- testcontainers + kotlin DSL + kotest extension (#1, #3, #4, #5)

### LINKED ISSUE #1: chore: bootstrap Gradle multi-module skeleton
## Context

Bootstrap the Gradle Kotlin DSL multi-module project that all subsequent modules (testcontainers, testcontainers-kotlin, testcontainers-kotest, later testcontainers-junit + spring-boot-starter) will live in. Pure scaffolding — no functional code yet.

## What to deliver

- `settings.gradle.kts` declaring the project name and including modules: `testcontainers`, `testcontainers-kotlin`, `testcontainers-kotest`. JUnit and Spring Boot modules added later in v0.2.0.
- Root `build.gradle.kts` with shared plugin config (Kotlin JVM, Java toolchain, version + group conventions).
- `gradle.properties`: `group=io.httptape`, `version=0.1.0-SNAPSHOT`, JVM toolchain target (25).
- Per-module `build.gradle.kts` for each of the 3 modules (initially empty source dirs; just enough to compile).
- Gradle wrapper committed (`gradlew`, `gradlew.bat`, `gradle/wrapper/`) — pin to the latest stable Gradle.
- `.gitignore` covering Gradle build artifacts + IntelliJ + JVM defaults.
- Replace placeholder `README.md` with a real one: project overview, modules table, quick-start install snippets (Maven + Gradle), link back to the main `httptape` repo.
- CI workflow stub: `.github/workflows/build.yml` running `./gradlew build` on PR + push to main.

## Acceptance criteria

1. `./gradlew build` runs cleanly with no errors (empty modules compile fine).
2. CI workflow runs successfully on push.
3. Module names in artifacts are: `httptape-testcontainers`, `httptape-testcontainers-kotlin`, `httptape-testcontainers-kotest` (Maven coordinates `io.httptape:httptape-*`). Gradle module directories strip the `httptape-` prefix per project convention.
4. README has install snippets for both Maven and Gradle.

## Out of scope

- Maven Central publishing config (separate issue: Maven Central publishing infrastructure).
- Functional code in any module (separate issues per module).
- Spring Boot / JUnit modules (v0.2.0).

## Dependencies

None. This is the foundation issue.

### LINKED ISSUE #3: feat(testcontainers): Java core — HttptapeContainer + fluent API
## Context

Implement the Java core module: `httptape-testcontainers`. A `HttptapeContainer` class extending Testcontainers' `GenericContainer<HttptapeContainer>` with a fluent Java API for the common httptape configuration patterns. This is the foundation that the Kotlin DSL and Kotest extension build on top of.

The goal: replace ~50 lines of `TestcontainersConfig.java` boilerplate (in the existing httptape Java demo) with ~5 lines of `HttptapeContainer` setup.

## API surface (initial)

```java
public class HttptapeContainer extends GenericContainer<HttptapeContainer> {
    // Default image: ghcr.io/vibewarden/httptape:0.13.1 (or latest)
    public HttptapeContainer() { ... }
    public HttptapeContainer(String image) { ... }
    
    // Fluent config — chain-friendly, returns this
    public HttptapeContainer withFixturesPath(Path hostPath);
    public HttptapeContainer withFixturesFromClasspath(String classpathRoot);
    public HttptapeContainer withMatcherConfig(Path configFile);
    public HttptapeContainer withMatcherConfigFromClasspath(String resourcePath);
    public HttptapeContainer withSseTiming(SseTimingMode mode);  // REALTIME, INSTANT, accelerated(factor)
    public HttptapeContainer withCors(boolean enabled);
    public HttptapeContainer withReplayHeader(String key, String value);
    public HttptapeContainer withDelay(Duration delay);
    public HttptapeContainer withErrorRate(double rate);
    public HttptapeContainer withFallbackStatus(int status);
    
    // Convenience accessors
    public String getBaseUrl();  // http://host:mappedPort
    public int getPort();
}
```

Includes an `SseTimingMode` value-object (sealed interface or enum + factory).

Image version pinned to a constant (`HttptapeContainer.DEFAULT_IMAGE = "ghcr.io/vibewarden/httptape:0.13.1"`) — bump on each httptape release. Users can override per-instance.

## What to deliver

- `testcontainers/src/main/java/dev/httptape/testcontainers/HttptapeContainer.java` (the main class).
- Supporting types: `SseTimingMode`, possibly `MatcherConfig` if classpath fixture loading needs structured representation.
- Comprehensive Javadoc on every public method.
- Integration tests using Testcontainers itself (test that container starts, `getBaseUrl()` works, fixtures are mountable, matcher config takes effect, etc.).
- Module-level README explaining install + minimal usage.

## Acceptance criteria

1. A Java integration test can replace ~50 LOC of the existing httptape Java demo's `TestcontainersConfig.java` with ~5 LOC using `HttptapeContainer`.
2. All public methods have Javadoc.
3. Tests pass via `./gradlew :testcontainers:test`.
4. Module produces a valid Maven artifact (`io.httptape:httptape-testcontainers:0.1.0-SNAPSHOT`).
5. Fixtures-from-classpath helper handles the same flat-mount pattern the Java demo uses today (avoid filename collisions).

## Open questions

1. Default image version: pin to `0.13.1` (current latest)? Pin to `latest`? PM lean: pin to a specific version per release of the SDK; bump SDK's default image version as part of each new SDK release. This couples SDK version to httptape version which is a feature, not a bug.
2. Classpath fixture loading: Spring's `PathMatchingResourcePatternResolver` is convenient but pulls Spring as a transitive dep. Use stdlib `ClassLoader.getResources()` instead? Architect's call.
3. SseTimingMode shape: enum with factory for `accelerated(factor)` vs sealed interface with three implementations. Sealed interface is more type-safe; enum is more familiar to Java users.

## Out of scope

- Kotlin DSL (separate module)
- Kotest extension (separate module)
- Spring Boot autoconfig (v0.2.0)

## Dependencies

Hard: bootstrap (#1) and publishing (#2) should land first so this can be built and (eventually) published.

### LINKED ISSUE #4: feat(testcontainers-kotlin): Kotlin DSL on top of the Java core
## Context

Implement the Kotlin DSL module: `httptape-testcontainers-kotlin`. Layers idiomatic Kotlin (extension functions, lambda receivers, scope functions) on top of the Java core (`httptape-testcontainers`). 

Goal: existing httptape Kotlin demo's `HttptapeContainer.kt` (~50 lines today) can be expressed in <10 lines using the DSL.

## API surface (initial)

```kotlin
// Top-level builder function with lambda receiver
fun httptape(image: String? = null, configure: HttptapeContainerScope.() -> Unit): HttptapeContainer

// Scope class hosting the DSL
class HttptapeContainerScope internal constructor(...) {
    fun fixtures(classpathRoot: String) { ... }
    fun fixturesPath(path: Path) { ... }
    fun matcherConfig(classpathPath: String) { ... }
    fun matcherConfig(path: Path) { ... }
    var sseTiming: SseTimingMode  // assignable property
    var cors: Boolean
    fun replayHeader(key: String, value: String) { ... }
    var delay: Duration?
    var errorRate: Double
    var fallbackStatus: Int
}

// Usage:
val container = httptape {
    fixtures("fixtures/")
    matcherConfig("httptape.config.json")
    sseTiming = SseTimingMode.Realtime
}
container.start()
```

Plus extension functions on the underlying `HttptapeContainer` for ad-hoc Kotlin-flavored access.

## What to deliver

- `testcontainers-kotlin/src/main/kotlin/dev/httptape/testcontainers/HttptapeDsl.kt` (top-level builder + scope class).
- Kotlin extension functions on `HttptapeContainer` for any methods that benefit from Kotlin idioms (e.g., `withDelay(100.milliseconds)` using kotlin.time).
- KDoc on all public API.
- Tests demonstrating the DSL replaces Java boilerplate.
- Module-level README with Kotlin install snippet.

## Acceptance criteria

1. The current httptape Kotlin demo's `HttptapeContainer.kt` (~50 LOC) can be re-expressed in <10 LOC using the DSL.
2. KDoc on all public API.
3. `./gradlew :testcontainers-kotlin:test` passes.
4. Module produces a valid Maven artifact (`io.httptape:httptape-testcontainers-kotlin:0.1.0-SNAPSHOT`).
5. Backward compat: code using the underlying Java `HttptapeContainer` directly continues to work — the DSL is purely additive.

## Open questions

1. Lambda receiver vs builder pattern: Kotlin's lambda receiver (`httptape { ... }`) is the canonical DSL shape. Confirm.
2. `SseTimingMode` exposure: re-export the Java type as-is, or re-define as a Kotlin sealed class for better type-safety? Re-export is simpler; sealed class is more idiomatic. Architect's call.
3. `kotlin.time.Duration` vs `java.time.Duration` for the `delay` setter: support both via overloads? Or pick one (probably kotlin.time for the DSL surface).

## Out of scope

- Kotest integration (separate module)
- Coroutine-aware lifecycle (probably not needed — Testcontainers itself is blocking)

## Dependencies

Hard: testcontainers Java core (#3). Cannot start until #3 lands or at least defines its public API.

### LINKED ISSUE #5: feat(testcontainers-kotest): Kotest extension on top of the Kotlin DSL
## Context

Implement the Kotest extension module: `httptape-testcontainers-kotest`. Layers a Kotest TestListener / extension on top of the Kotlin DSL (`httptape-testcontainers-kotlin`). Manages container lifecycle (start before, stop after) at the spec or test-case scope.

Goal: existing httptape Kotlin demo's `WeatherAdviceTest` can use Kotest's `install()` pattern instead of the singleton `HttptapeContainer object` pattern.

## API surface (initial)

```kotlin
// Kotest extension factory function
fun httptapeExtension(configure: HttptapeContainerScope.() -> Unit): HttptapeExtension

class HttptapeExtension internal constructor(...) : TestListener {
    val baseUrl: String  // exposed for tests to read
    val container: HttptapeContainer
    
    override suspend fun beforeSpec(spec: Spec) { container.start() }
    override suspend fun afterSpec(spec: Spec) { container.stop() }
}

// Usage in a Kotest spec:
class WeatherAdviceTest : FreeSpec({
    val httptape = install(httptapeExtension {
        fixtures("fixtures/")
        matcherConfig("httptape.config.json")
    })
    
    "advises bringing an umbrella when it is rainy in the requested city" {
        // Use httptape.baseUrl in the test
    }
})
```

Lifecycle scope: per-spec (default) — one container shared across all test cases in the spec. Per-test scope is possible but expensive; defer to a follow-up if needed.

## What to deliver

- `testcontainers-kotest/src/main/kotlin/dev/httptape/testcontainers/HttptapeExtension.kt`
- KDoc on all public API
- Tests using Kotest itself (extension behavior verification)
- Module-level README with Kotest install snippet

## Acceptance criteria

1. The current httptape Kotlin demo's `WeatherAdviceTest` can use `install(httptapeExtension { ... })` instead of the singleton object.
2. Container lifecycle is correctly managed per-spec.
3. KDoc on all public API.
4. `./gradlew :testcontainers-kotest:test` passes.
5. Module produces a valid Maven artifact (`io.httptape:httptape-testcontainers-kotest:0.1.0-SNAPSHOT`).
6. Reuses the Kotlin DSL from `httptape-testcontainers-kotlin` — no duplicated configuration surface.

## Open questions

1. Kotest version baseline: Kotest 6.x latest stable. Confirm at impl time.
2. Extension type: `TestListener` (legacy, simpler) vs `Extension` (modern Kotest extension framework). Modern is preferred but verify ergonomics.
3. Scope: per-spec only, or expose `beforeTest`/`afterTest` for per-test isolation too? Defer per-test to a follow-up.
4. Coroutine context: Kotest is coroutine-native; container start/stop is blocking. Wrap in `withContext(Dispatchers.IO)` or let Kotest's default scheduler handle it? Architect's call.

## Out of scope

- Per-test container scope (defer)
- JUnit 5 extension (separate module, v0.2.0)

## Dependencies

Hard: testcontainers-kotlin (#4). Cannot start until #4 lands.

### PR DESCRIPTION
**Pull request by tibtof**
Closes #1, closes #3, closes #4, closes #5

## Summary

This PR delivers the complete v0.1 httptape-jvm SDK as a Gradle multi-module project with three modules:

- **`httptape-testcontainers`** (Java core): `HttptapeContainer` extending `GenericContainer` with a fluent API for fixtures, matcher config, SSE timing, CORS, delay, error rate, fallback status, and replay headers. `SseTimingMode` as a Java sealed interface with three record implementations. Classpath fixture scanning via stdlib `ClassLoader.getResources()` + `Files.walk()` -- no Spring dependency.

- **`httptape-testcontainers-kotlin`** (Kotlin DSL): `httptape { ... }` builder with `HttptapeContainerScope` DSL receiver. Re-exports Java `SseTimingMode` directly. Extension properties (`baseUrl`) and functions (`withDelay` accepting `kotlin.time.Duration`).

- **`httptape-testcontainers-kotest`** (Kotest extension): `httptapeExtension { ... }` factory with `HttptapeExtension` implementing `MountableExtension<Unit, HttptapeHandle>` for per-spec container lifecycle via Kotest's `install()` pattern.

### Design decisions

- Follows ADR-001 in `decisions.md`
- All four issues (#1 bootstrap, #3 Java core, #4 Kotlin DSL, #5 Kotest extension) bundled per architect's PR strategy
- Version catalog verified: Kotlin 2.3.20, Gradle 9.4.1, Testcontainers 2.0.4, Kotest 6.1.11, JUnit 5.14.3
- JDK 21 toolchain
- `maven-publish` included for `publishToMavenLocal` only (no remote repository config)

### Version note

The version catalog uses `kotest-framework-engine` instead of the ADR's `kotest-framework-api` because `kotest-framework-api` is no longer published as a separate artifact in Kotest 6.x -- the API types are bundled within `kotest-framework-engine`. The `kotest-runner-junit5` already depends on this transitively.

## Test plan

- [x] `./gradlew clean build` passes (47 tests across 3 modules, 0 failures)
- [x] `./gradlew publishToMavenLocal` publishes all three artifacts
- [ ] PR 2 in `VibeWarden/httptape` will refactor the existing demos to consume this SDK via composite build -- that is the validation gate

Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: testcontainers/src/main/java/dev/httptape/testcontainers/HttptapeContainer.java [hunk 1]
```diff
@@ -0,0 +1,419 @@
...
+
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.utility.MountableFile;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.file.FileSystem;
```
Comment (tibtof) [inline testcontainers/src/main/java/dev/httptape/testcontainers/HttptapeContainer.java]: **Unused imports**: `java.nio.file.FileSystem`, `java.nio.file.FileSystems`, and `java.util.Collections` are imported but never used in this file. These are likely leftovers from an earlier implementation approach.

Remove lines 11, 12, and 17 to clean up.

File: testcontainers/src/main/java/dev/httptape/testcontainers/SseTimingMode.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+            return "instant";
+        }
+    }
+
+    /**
+     * Replay at an accelerated speed.
+     *
+     * @param factor the acceleration multiplier (must be positive)
+     */
+    record Accelerated(double factor) implements SseTimingMode {
```
Comment (tibtof) [inline testcontainers/src/main/java/dev/httptape/testcontainers/SseTimingMode.java:69]: **Validation bypass**: The `Accelerated` record constructor is implicitly public (records inside interfaces are `public static` by default in Java). This means callers can write `new SseTimingMode.Accelerated(-1.0)` and bypass the validation in the `accelerated(double)` factory method.

Add a compact constructor with the same validation:

```java
record Accelerated(double factor) implements SseTimingMode {
    Accelerated {
        if (factor <= 0) {
            throw new IllegalArgumentException(
                "Acceleration factor must be positive, got: " + factor);
        }
    }
    @Override
    public String toCliFlag() {
        return "accelerated=" + factor;
    }
}
```

This ensures the invariant holds regardless of how the record is constructed.

File: gradle/libs.versions.toml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+junit-platform = "5.14.3"
+
+[libraries]
+testcontainers-core = { module = "org.testcontainers:testcontainers", version.ref = "testcontainers" }
+kotest-runner = { module = "io.kotest:kotest-runner-junit5", version.ref = "kotest" }
+kotest-assertions = { module = "io.kotest:kotest-assertions-core", version.ref = "kotest" }
+kotest-framework-engine = { module = "io.kotest:kotest-framework-engine", version.ref = "kotest" }
+junit-jupiter-api = { module = "org.junit.jupiter:junit-jupiter-api", version.ref = "junit-platform" }
+junit-jupiter-engine = { module = "org.junit.jupiter:junit-jupiter-engine", version.ref = "junit-platform" }
+junit-platform-launcher = { module = "org.junit.platform:junit-platform-launcher" }
```
Comment (tibtof) [inline gradle/libs.versions.toml]: **Missing version for `junit-platform-launcher`**: This catalog entry has no `version.ref`, which means the version will be resolved transitively. While this works today because JUnit Jupiter Engine pulls in a compatible platform version, it is fragile -- a future dependency update could break the resolution.

Pin it explicitly:

```toml
junit-platform-launcher = { module = "org.junit.platform:junit-platform-launcher", version.ref = "junit-platform" }
```

This keeps all JUnit components at the same `5.14.3` version.

File: testcontainers/src/main/java/dev/httptape/testcontainers/HttptapeContainer.java [hunk 4]
```diff
@@ -0,0 +1,419 @@
...
+    }
+
+    /**
+     * Mounts a host directory as the fixture source.
+     *
+     * @param hostPath path to the fixtures directory on the host
+     * @return this container for chaining
+     */
+    public HttptapeContainer withFixturesPath(Path hostPath) {
+        withFileSystemBind(hostPath.toAbsolutePath().toString(), CONTAINER_FIXTURES_PATH);
```
Comment (tibtof) [inline testcontainers/src/main/java/dev/httptape/testcontainers/HttptapeContainer.java]: **Nit (non-blocking)**: `withFileSystemBind` defaults to `BindMode.READ_WRITE`. Since fixtures are read-only data consumed by the container, consider using the three-argument form with `BindMode.READ_ONLY`:

```java
withFileSystemBind(hostPath.toAbsolutePath().toString(), CONTAINER_FIXTURES_PATH, BindMode.READ_ONLY);
```

This makes the intent explicit and prevents accidental writes to the host fixture directory.

**Review by tibtof — COMMENTED** [review]
**APPROVED** (posted as comment because GitHub prevents self-approval)

LGTM with minor fixes. Thorough review of all 5 commits across 24 files.

## Summary

This is a well-executed greenfield SDK. The three-module architecture (Java core -> Kotlin DSL -> Kotest extension) cleanly layers functionality with proper dependency direction. The code is idiomatic, well-documented, and the test suite is comprehensive (47 tests: 31 Java unit, 11 Kotlin DSL unit, 5 Kotest unit).

## Inline comments (4 total)

1. **Unused imports** in `HttptapeContainer.java` (lines 11, 12, 17): `FileSystem`, `FileSystems`, `Collections` -- remove.
2. **Accelerated record validation bypass** in `SseTimingMode.java` (line 69): The record's public constructor allows bypassing the factory method's validation. Add a compact constructor.
3. **Missing version** for `junit-platform-launcher` in `libs.versions.toml` (line 14): Pin to `junit-platform` version ref.
4. **Nit**: `withFixturesPath` should use `BindMode.READ_ONLY` for fixture mounts.

None of these are blocking -- they are all quick fixes.

## Detailed review findings

### (a) ADR-001 fidelity
Implementation matches ADR-001 faithfully. All locked constants are correct:
- Maven group: `dev.httptape`
- Default image: `ghcr.io/vibewarden/httptape:0.13.1`
- Container port: 8081
- Fixture path: `/fixtures`
- Config path: `/config/httptape.config.json`
- JDK toolchain: 21
- All dependency versions match (Kotlin 2.3.20, Gradle 9.4.1, Testcontainers 2.0.4, Kotest 6.1.11, JUnit 5.14.3)

The `kotest-framework-api` -> `kotest-framework-engine` deviation is correct for Kotest 6.x which merged the API artifact into the engine. ADR should be updated to reflect this, but it is not blocking.

### (b) Idiomatic Kotlin/Java
- **Java**: Proper use of sealed interface + records for `SseTimingMode`. Fluent API with `self()` returns. No `var` overuse. Javadoc on all public methods.
- **Kotlin DSL**: Idiomatic `@DslMarker` annotation. Lambda receiver pattern. `kotlin.time.Duration` for delay. Property delegates with custom setters. Extension functions for `baseUrl` and `withDelay`.
- **Kotest extension**: Correct use of `MountableExtension<Unit, HttptapeHandle>` with `BeforeSpecListener`/`AfterSpecListener`. No deprecated `TestListener`.

### (c) Test quality
Strong mix of unit and integration-flavored tests:
- SseTimingMode: 7 tests covering all variants, edge cases (zero, negative), and sealed interface verification
- Command assembly: 13 tests covering every CLI flag combination
- Classpath fixture discovery: 3 tests (filesystem walk, collision detection, missing root)
- Error rate validation: 4 tests (boundary values)
- Fluent API: 1 test asserting `assertSame` on all fluent methods
- Default image: 3 tests
- Kotlin DSL: 11 tests mirroring Java command assembly through the DSL layer
- Kotest extension: 5 tests (factory, mount, handle, image override)

No bare `assertTrue(thing != null)` -- assertions are meaningful throughout.

### (d) Kotest 6 / Testcontainers 2 API usage
- `MountableExtension` is the correct Kotest 6 pattern
- `BeforeSpecListener` / `AfterSpecListener` are current interfaces (not the deprecated `TestListener`)
- `GenericContainer<HttptapeContainer>` self-typing is correct for Testcontainers 2.x
- `Wait.forHttp("/").forStatusCode(404)` is a valid wait strategy

### (e) Public API surface for PR 2
The API matches what the demos will consume:
- `HttptapeContainer`: all 10 fluent methods + `getBaseUrl()` + `getPort()` + `buildCommand()` (extra, for testing)
- `SseTimingMode`: sealed interface with `REALTIME`, `INSTANT`, `accelerated(double)`
- Kotlin DSL: `httptape { ... }` builder, `HttptapeContainerScope` with all properties/functions
- Kotest: `httptapeExtension { ... }` factory, `HttptapeExtension`, `HttptapeHandle` with `baseUrl` and `port`

The composite build snippet in the README matches what PR 2 will use in the demo's `settings.gradle.kts`.

### Additional observations (non-blocking, not commented inline)
- `maven-publish` is correctly configured for `publishToMavenLocal` only -- no remote repository URLs anywhere.
- All READMEs note "v0.1.0 is not yet published to Maven Central."
- `.gitignore` covers Gradle, IntelliJ, JVM, and `local-repo/`.
- CI workflow is minimal and correct (checkout, setup-java 21, setup-gradle, build, upload reports on failure).
- Default image pin is a single constant (`DEFAULT_IMAGE`) -- bumping requires one line change.
- No TODO comments in production code.
- No secrets or credentials hardcoded.

File: testcontainers/src/main/java/dev/httptape/testcontainers/HttptapeContainer.java (hunk 1 above)
Comment (tibtof) [reply testcontainers/src/main/java/dev/httptape/testcontainers/HttptapeContainer.java]: Fixed in 022742f. Removed all three unused imports (FileSystem, FileSystems, Collections).

File: testcontainers/src/main/java/dev/httptape/testcontainers/SseTimingMode.java (hunk 2 above)
Comment (tibtof) [reply testcontainers/src/main/java/dev/httptape/testcontainers/SseTimingMode.java:69]: Fixed in 022742f. Added a compact constructor to the Accelerated record that validates factor > 0, so both the factory method and direct instantiation are now guarded. Also added two unit tests (acceleratedRecord_zeroFactor_throws, acceleratedRecord_negativeFactor_throws) to cover the constructor path.

File: gradle/libs.versions.toml (hunk 3 above)
Comment (tibtof) [reply gradle/libs.versions.toml]: Fixed in 022742f. Split the version catalog into junit-jupiter = "5.14.3" (for org.junit.jupiter artifacts) and junit-platform = "1.14.3" (for org.junit.platform artifacts). The junit-platform-launcher now uses version.ref = "junit-platform" pointing to 1.14.3.

File: testcontainers/src/main/java/dev/httptape/testcontainers/HttptapeContainer.java (hunk 4 above)
Comment (tibtof) [reply testcontainers/src/main/java/dev/httptape/testcontainers/HttptapeContainer.java]: Fixed in 022742f. withFixturesPath now explicitly passes BindMode.READ_ONLY. The other mount points (withFixturesFromClasspath, withMatcherConfig, withMatcherConfigFromClasspath) use withCopyFileToContainer which copies files into the container filesystem -- BindMode does not apply there.
